app/models/person.rb

```diff
@@ -0,0 +1,11 @@
+class Person < ActiveRecord::Base
+  has_many :tasks
+
+  def delete_associated_tasks
+    Task.where(person_id: self.id).destroy_all
+  end
+
+  def tasks_remaining
+    Task.where("person_id = ? AND completed_at IS NULL", self.id ).count
```

Comment:
if you use the `tasks` as I suggested above you could move the `completed_at IS NULL` sql into the `Task` model, which seems like a better place for knowledge about `Task` internal logic. They you'd just call `tasks.remaining.count` or something similar